poly-commitment/src/commitment.rs

```diff
@@ -33,11 +33,9 @@ use super::evaluation_proof::*;
 
 /// A polynomial commitment.
 #[serde_as]
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct PolyComm<C>
-where
-    C: CanonicalDeserialize + CanonicalSerialize,
-{
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+#[serde(bound = "C: CanonicalDeserialize + CanonicalSerialize")]
```

Comment:
ah interesting, IIUC this will do something like

```rust
impl<C> serde::Serialize for PolyComm<C> where C: CanonicalSerialize + CanonicalDeserialize { ... }
```

and as such as long as we don't (de)serialize the struct the bound won't matter. I guess we should do that for all of our structs then : o (we often just add `F: (Prime)Field` to satisfy serde).
```